src/test/java/io/jenkins/plugins/junit/checks/JUnitChecksPublisherTest.java

```diff
@@ -2,29 +2,40 @@
 
 import hudson.FilePath;
 import hudson.model.Result;
-import hudson.tasks.junit.TestResultAction;
-import hudson.tasks.junit.TestResultSummary;
 import hudson.tasks.junit.TestResultTest;
 import io.jenkins.plugins.checks.api.ChecksConclusion;
 import io.jenkins.plugins.checks.api.ChecksDetails;
 import io.jenkins.plugins.checks.api.ChecksOutput;
+import io.jenkins.plugins.checks.api.ChecksStatus;
 import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
 import org.jenkinsci.plugins.workflow.job.WorkflowJob;
-import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 import org.jvnet.hudson.test.JenkinsRule;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
 import static java.util.Objects.requireNonNull;
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.is;
-import static org.junit.Assert.assertNotNull;
+import static org.mockito.ArgumentMatchers.*;
+import static org.powermock.api.mockito.PowerMockito.*;
 
+@RunWith(PowerMockRunner.class)
```

Comment:
Beware that every time I have come across Mockito or PowerMock tests in Jenkins core or plugins, it has ended in disaster. They tend to require way more maintenance than `JenkinsRule` tests (with `@TestExtension` mocks as needed), and fail to detect real regressions.

Reply:
It has been fun getting them to work (mostly because my work machine decided to OOM every other run...) but I couldn't see another obvious point at which to intercept this _without_ some kind of mocks. I'm certainly not tied to the current implementation, but we have improved coverage this way at least.

Reply:
> obvious point at which to intercept this

Not sure I am following. https://github.com/jenkinsci/junit-plugin/blob/91f112be2b2b7f9bc056eb7ee0bb4f6ab7d645c2/src/test/java/io/jenkins/plugins/junit/checks/JUnitChecksPublisherTest.java#L32-L40 and the like already run actual builds with stages and actual test results, and https://github.com/jenkinsci/junit-plugin/blob/91f112be2b2b7f9bc056eb7ee0bb4f6ab7d645c2/src/test/java/io/jenkins/plugins/junit/checks/JUnitChecksPublisherTest.java#L51 already asserts the check name.

Reply:
That only tests the constructor of `JUnitChecksPublisher`; without this change none of the code in `JUnitResultArchiver#parseAndSummarize` is tested, which is where we have access to the task config and pipline test details. See for example 
https://github.com/jenkinsci/junit-plugin/blob/91f112be2b2b7f9bc056eb7ee0bb4f6ab7d645c2/src/test/java/io/jenkins/plugins/junit/checks/JUnitChecksPublisherTest.java#L138
Where we can't (currently) test that the custom check name is correctly extracted. 

This new way of testing would also allow us to write tests for the bug in https://github.com/jenkinsci/junit-plugin/pull/199, which again was code in `parseAndSummarize` that did not have coverage.

Reply:
you could extract that code out to somewhere easier to test, either in the checks publisher or somewhere else?

Reply:
I'll have a play :)

Reply:
Ahh, thanks for the hint on `@TestExntesion`, that's much neater.

Apologies if my implementation here isn't terribly idiomatic; if it's not already obvious my Java is a bit rusty and I'm not really up to date with the best ways of doing things.

Reply:
I am afraid Jenkins programming has little to do with general Java experience; it is very idiosyncratic due to the old bespoke frameworks.